File: AbstractFactoryPattern/Java/FurnitureStore/src/com/laxmena/factory/FurnitureFactory.java
package com.laxmena.factory;

import com.laxmena.chair.Chair;
import com.laxmena.sofa.Sofa;
import com.laxmena.table.Table;

public interface FurnitureFactory {
    public Chair createChair();
    public Sofa createSofa();
    public Table createTable();
}

File: AbstractFactoryPattern/Python/FurnitureStore/factory/furniture_factory.py
from abc import ABC, abstractmethod

from products.chair.chair import Chair
from products.sofa import Sofa
from products.table import Table


class FurnitureFactory(ABC):
    @abstractmethod
    def create_chair(self) -> Chair:
        pass

    @abstractmethod
    def create_sofa(self) -> Sofa:
        pass

    @abstractmethod
    def create_table(self) -> Table:
        pass
